teachertool/src/components/SplitPane.tsx

```diff
@@ -5,20 +5,115 @@ import { classList } from "react-common/components/util";
 interface IProps {
     className?: string;
     split: "horizontal" | "vertical";
-    defaultSize: number | string;
+    defaultSize: number | string; // The size to reset to when double clicking the splitter.
+    startingSize?: number | string; // The size to use initially when creating the splitter. Defaults to `defaultSize`.
```

Comment:
I'm not sure I understand the benefit of having this extra parameter. I can understand that the parent of the `SplitPane` should be able to declare the default size of the panes, but I think starting size is something that can be determined inside `SplitPane`.

I think just having `defaultSize` is all we need, and then when first setting the size in the `useState`, you can just have the `getLastSplitPosition` call as the `lastSavedPosition` variable and the `useState` are essentially doing the same thing right now.

Reply:
Oh, I see. You're trying to keep the state/service stuff outside of `SplitPane` to make it easier to move in the future. I wonder if saving the size in localstorage is something you just want to bring with this, though.

Reply:
I don't think we should have this class interface with local storage. The caller would still need to pass in the local storage key to use, which would feel a little weird, but more importantly, if we re-use it somewhere else, that behavior may not be desirable. We may not want to save it at all, or we may want to save it somewhere else, like an account-level setting for example.